Drv/LinuxI2cDriver/LinuxI2cDriver.cpp

```diff
@@ -174,25 +174,23 @@ namespace Drv {
       Fw::Logger::logMsg("I2c addr: 0x%02X\n",addr);
     #endif
 
-    struct i2c_msg rdwr_msgs[2] = {
-        {  // Start address
-            .addr = static_cast<U16>(addr),
-            .flags = 0, // write
-            .len = static_cast<U16>(writeBuffer.getSize()),
-            .buf = writeBuffer.getData()
-        },
-        { // Read buffer
-            .addr = static_cast<U16>(addr),
-            .flags = I2C_M_RD, // read
-            .len = static_cast<U16>(readBuffer.getSize()),
-            .buf = readBuffer.getData()
-        }
-    };
-
-    struct i2c_rdwr_ioctl_data rdwr_data = {
-        .msgs = rdwr_msgs,
-        .nmsgs = 2
-    };
+    struct i2c_msg rdwr_msgs[2];
```

Comment:
I don't understand why this needed to be refactored.  We use this style initialization in multiple places.  Is this a violation of `-pedantic`?

Reply:
From what I have seen, it has only been officially supported by c++20, before that it was just an extension of GCC, but not part of the official c++11 standard, hence the pedantic warning.

```
error: C++ designated initializers only available with ‘-std=c++2a’ or ‘-std=gnu++2a’ [-Werror=pedantic]
```

I checked the code base using some regex to look for this type of initialization, and I did not spot any other places, other than in the shared folder for some tests.

Reply:
Thanks for looking into this!  I will re-review!